independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/StringUtil.java

```diff
@@ -0,0 +1,104 @@
+                        if (Character.isUpperCase(c)) {
+                            // all-caps word; need one look-ahead
+                            int nextIdx = str.offsetByCodePoints(idx, 1);
+                            while (nextIdx < str.length()) {
+                                c = str.codePointAt(nextIdx);
+                                if (Character.isLowerCase(c)) {
+                                    // ended at idx
+                                    return str.substring(start, idx);
+                                }
+                                idx = nextIdx;
+                                nextIdx = str.offsetByCodePoints(idx, 1);
+                            }
+                            // consumed the whole remainder, update idx to length
+                            idx = str.length();
+                            return str.substring(start);
+                        } else {
+                            // initial caps, trailing lowercase
+                            idx = str.offsetByCodePoints(idx, 1);
+                            while (idx < str.length()) {
+                                c = str.codePointAt(idx);
+                                if (Character.isUpperCase(c)) {
+                                    // end
+                                    return str.substring(start, idx);
+                                }
+                                idx = str.offsetByCodePoints(idx, 1);
+                            }
+                            // consumed the whole remainder
+                            return str.substring(start);
+                        }
+                    } else {
+                        // one-letter word
+                        return str.substring(start);
+                    }
+                } else {
+                    // a lowercase-starting word
+                    idx = str.offsetByCodePoints(idx, 1);
+                    while (idx < str.length()) {
+                        c = str.codePointAt(idx);
+                        if (Character.isUpperCase(c)) {
+                            // end
+                            return str.substring(start, idx);
+                        }
+                        idx = str.offsetByCodePoints(idx, 1);
+                    }
+                    // consumed the whole remainder
+                    return str.substring(start);
+                }
+            }
+        };
+    }
+
+    /**
+     * @deprecated Use {@link String#join} instead.
+     * @param delim delimiter
+     * @param it iterator
+     * @return the joined string
+     */
+    @Deprecated
+    public static String join(final String delim, final Iterator<String> it) {
+        final StringBuilder b = new StringBuilder();
+        if (it.hasNext()) {
+            b.append(it.next());
+            while (it.hasNext()) {
+                b.append(delim);
+                b.append(it.next());
+            }
+        }
+        return b.toString();
+    }
```

Comment:
Remove it anyway? I don't think anyone is using it TBH

Reply:
I added this because [`String.join`](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#join-java.lang.CharSequence-java.lang.Iterable-) receives an `Iterable` and `camelHumpsIterator` method return an `Iterator` but I think it's better to cast the return from `camelHumpsIterator` to an iterable and use `String.join`.